File: 0001_0100/0094/0094.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m, k = (int(x) for x in f.read().split())
    q.write("1" if n >= m else "NO" if n <= k else str((m - n - 1) // (n - k) + 2))

File: 0001_0100/0094/0094.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, k;

	fscanf(f, "%d%d%d", &n, &m, &k);
	if (n <= k && n < m)
		fprintf(q, "NO");
	else fprintf(q, "%d", n >= m ? 1 : (m - n - 1) / (n - k) + 2);
	return 0;
}
